larray/viewer.py

```diff
@@ -1949,44 +2088,24 @@ def get_value(self):
 DISPLAY_IN_GRID = (la.LArray, np.ndarray)
 
 
-class MappingEditor(QDialog):
+class MappingEditor(AbstractEditor):
     """Session Editor Dialog"""
     def __init__(self, parent=None):
-        QDialog.__init__(self, parent)
-
-        # Destroying the C++ object right after closing the dialog box,
-        # otherwise it may be garbage-collected in another QThread
-        # (e.g. the editor's analysis thread in Spyder), thus leading to
-        # a segmentation fault on UNIX or an application crash on Windows
-        self.setAttribute(Qt.WA_DeleteOnClose)
+        AbstractEditor.__init__(self, parent)
 
-        self.data = None
         self._listwidget = None
-        self.arraywidget = None
         self.eval_box = None
         self.expressions = {}
         self.kernel = None
 
-    def setup_and_check(self, data, title='', readonly=False,
-                        minvalue=None, maxvalue=None):
+    def setup_and_check(self, data, title='', readonly=False, minvalue=None, maxvalue=None):
         """
         Setup MappingEditor:
         return False if data is not supported, True otherwise
         """
-        self.data = data
-
-        icon = self.style().standardIcon(QStyle.SP_ComputerIcon)
-        if icon is not None:
-            self.setWindowIcon(icon)
-
-        if not title:
-            title = _("Session viewer") if readonly else _("Session editor")
-        if readonly:
-            title += ' (' + _('read only') + ')'
-        self.setWindowTitle(title)
-
-        layout = QVBoxLayout()
-        self.setLayout(layout)
+        if not isinstance(data, la.Session):
+            data = la.Session(data)
```

Comment:
Unsure we want this. This might make using the viewer as a debugger harder. We need to this think through.